manual_refresh_token_provider.go

```diff
@@ -0,0 +1,33 @@
+package incognia
+
+import "errors"
+
+var ErrNoToken = errors.New("no token stored in memory")
+
+type ManualRefreshTokenProvider struct {
+	tokenClient *TokenClient
+	token       *AccessToken
+}
+
+func NewManualRefreshTokenProvider(tokenClient *TokenClient) *ManualRefreshTokenProvider {
+	return &ManualRefreshTokenProvider{tokenClient: tokenClient}
+}
+
+func (t *ManualRefreshTokenProvider) GetToken() (*AccessToken, error) {
```

Comment:
I would return and error when the token is expired, to avoid clients making requests that we already know will fail.